packages/wallets/src/components/WalletsCarousel/WalletsCarousel.tsx

```diff
@@ -9,57 +8,54 @@ import './WalletsCarousel.scss';
 
 const WalletsCarousel: React.FC = () => {
     const [isWalletSettled, setIsWalletSettled] = useState(true);
-    const [showWalletsCarouselHeader, setShowWalletsCarouselHeader] = useState(false);
-    const [heightFromTop, setHeightFromTop] = useState(0);
+    const [hideWalletsCarouselHeader, setHideWalletsCarouselHeader] = useState(false);
+    const contentRef = useRef(null);
     const { data: activeWallet, isLoading: isActiveWalletLoading } = useActiveWalletAccount();
 
-    const containerRef = useRef<HTMLDivElement>(null);
-
-    // function to handle scrolling event for hiding/displaying WalletsCarouselHeader
-    // walletsCarouselHeader will be displayed when height from top of screen is more than 100px
-    const handleScroll = useCallback(() => {
-        if (containerRef.current) {
-            const newHeightFromTop = containerRef.current.getBoundingClientRect().top;
-            setHeightFromTop(newHeightFromTop);
-            heightFromTop && setShowWalletsCarouselHeader(heightFromTop < -100);
-        }
-    }, [heightFromTop]);
-
-    //this handle scroll function listens to the scroll as well as touchmove events to handle drag scrolling on mobile
-    useEventListener('touchmove', handleScroll, containerRef);
-    useEventListener('touchend', handleScroll, containerRef);
-    useEventListener('scroll', handleScroll, containerRef);
-    useEventListener('wheel', handleScroll, containerRef);
-
+    // useEffect hook to handle event for hiding/displaying WalletsCarouselHeader
+    // walletsCarouselHeader will be displayed when WalletsCarouselContent is almost out of viewport
     useEffect(() => {
-        let isMounted = true;
+        const observer = new IntersectionObserver(
+            entries => {
+                entries.forEach(entry => {
+                    setHideWalletsCarouselHeader(entry.isIntersecting);
+                });
+            },
+            { threshold: 0.4 } // triggers when 40% of the element is left in view
+        );
+
+        const currentContentRef = contentRef.current;
 
-        if (isMounted) {
-            handleScroll();
+        if (currentContentRef) {
+            observer.observe(currentContentRef);
         }
 
         return () => {
-            isMounted = false;
+            if (currentContentRef) {
+                observer.unobserve(currentContentRef);
+            }
         };
-    }, [handleScroll, heightFromTop]);
+    }, []);
```

Comment:
Hi @lubega-deriv,
Could we use the [usehooks-ts](https://usehooks-ts.com/react-hook/use-intersection-observer) implementation instead of doing it manually?

Reply:
@aum-deriv tried it out with useIntersectionObserver but seems like we are not using the latest version of usehooks-ts in wallets and the implementation has changed/is deprecated. We can update the version but we'll have to find alternatives for the removed hooks used inside wallets such as useDebounce
<img width="731" alt="image" src="https://github.com/binary-com/deriv-app/assets/142860499/7f3d3ca1-82c4-4a78-8afd-1a43d714e465">

Reply:
Alright, then for now we can proceed with this implementation. Later we can think of using it when we update the library 👍